app/services/sign_up_service.rb

```diff
@@ -0,0 +1,59 @@
+class SignUpService
+
+  attr_accessor :user, :status, :message
+
+  def initialize(user)
+    @user = user
+    @status = status
+    @message = message
+  end
+
+  def signup(invitation_token, stripe_token)
```

Comment:
I would set `invitation_token` to default `nil`. Its not always sign up will be with token.